doc/src/bond_over_grpc.md

```diff
@@ -90,21 +96,141 @@ On the client side, the proxy stub establishes a connection to the server like t
 The proxy stub can then be used to make calls to the server as follows:
 
     var request = new ExampleRequest();
+    // Fill in request fields here
+
     IMessage<ExampleResponse> responseMessage = await client.Method(request);
-    var response = responseMessage.Payload.Deserialize().Payload;
 
-For more information about gRPC in C#, take a look at the
-[gRPC C# tutorial](http://www.grpc.io/docs/tutorials/basic/csharp.html).
+    ExampleResponse response = responseMessage.Payload.Deserialize().Payload;
+    // Examine response here
+
 Note that the singatures generated by `gbc` are slightly different from the
-ones in the gRPC tutorial: on the service side, the request is wrapped in
+ones in the gRPC documentation: on the service side, the request is wrapped in
 `IMessage<T>` and on the client side, the response is wrapped in
-`IMessage<T>`. This allows for better control over the time of
+`IMessage<T>; this allows for better control over the time of
 deserialization and also helps prevent slicing when using polymorphic Bond
 types. Note also that Bond-over-gRPC does not provide synchronous APIs in C#
 by design.
 
+For more information about gRPC in C#, take a look at the
+[gRPC C# tutorial](http://www.grpc.io/docs/tutorials/basic/csharp.html).
+
 There is a [Bond-over-gRPC standalone example project](https://github.com/Microsoft/bond-grpc-examples).
 
 See also the following example:
 
 - `examples/cs/grpc/pingpong`
+
+## Bond-over-gRPC for C++ ##
+
+Given a service definition like the following:
+
+    service Example
+    {
+        ExampleResponse ExampleMethod(ExampleRequest);
+    }
+
+`gbc` will generate C++ classes for gRPC with the `--grpc` flag:
+
+    gbc c++ --grpc example.bond
+
+The key generated C++ classes for gRPC are:
+* A simple class with the name of the service (e.g.: `Example`). This class
+  provides some basic encapsulation of the server-side service base and the
+  client-side proxy stub.
+* The service base, which is an inner class named `Service` (e.g.:
+  `Example::Service`). [^1] This class has abstract methods
```

Comment:
Just so you're aware, this footnote syntax is Pandoc-specific. GitHub, for example, just has "\[^1\]" in the paragraph here and odd text at the bottom of the page.

(No change needed based on this comment, unless you want to rework the prose.)